modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/ResolverSetResult.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.extension.introspection.Parameter;
+
+import com.google.common.base.Objects;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * This class represents the outcome of the evaluation of a {@link ResolverSet}.
+ * This class maps a set of {@link Parameter} to a set of result {@link Object}s.
+ * <p/>
+ * This classes {@link #equals(Object)} and {@link #hashCode()} methods have been redefined
+ * to be consistent with the result objects. This is so that given two instances of this class
+ * you can determine if the evaluations they represent have an equivalent outcome
+ * <p/>
+ * Instances of this class are immutable and can only be created through a {@link Builder}
```

Comment:
Instances are partially immutable, as values assigned to parameters can be mutable